DanielAdek/Finance-Palace-API
# #chore: Setup project

### What does this PR do?
Setup a Node TypeScript Express application.

### Prerequisite
To test this PR you should have the following:
* You should have a code editor (e.g [vscode)](https://code.visualstudio.com) on your machine
* You should have installed git on your system
* You should have [mongodb](https://www.mongodb.com/download-center/enterprise) installed on your system
* You should have installed [postman](https://www.getpostman.com/downloads/) or [Insomnia](https://insomnia.rest).

### How do I test this PR?
* Clone this repo,
* Open your terminal and locate the project directory, and run `git checkout ch-setup-project`,
* Run `yarn install`,
* Follow the example in the `.env.example` file, and create a `.env` as-is,
* On another terminal run `mongod` to start mongo database,
* Run `yarn build-ts && yarn start` to start the app or `yarn build-ts && yarn watch-node` to start the app with nodemon or `yarn watch` to start the app node and typescript in watch mode
* On postman or insomnia, hit the following endpoints to perform crud operation.

  * The base url is: `http://localhost:${your-port}/api/v1`.
  - A `get` request on `/` to see server response.

### Screenshots
_Sample of success response when the server is up and running below:_
![Screenshot 2020-01-04 at 2 21 01 AM](https://user-images.githubusercontent.com/36076218/71757805-268efb00-2e99-11ea-9998-c6ab29cbd736.png)
